### ISSUE TITLE
Installation issue: py-tensorflow on apollo80 with gcc 

### ISSUE THREAD

**Issue by samcom12** [LLNL/spack#21190]
I have tried building py-TensorFlow with GCC on the apollo80 system. but it was failing for bazel so I installed bazel manually and added same to ```packages.yaml```

```
bazel:
    externals:
    - spec: bazel@3.7.1
      prefix: /home/apps/bazel_371
```

### Steps to reproduce the issue

```console
$ spack install  py-tensorflow+mpi~cuda~nccl %gcc ^mpich ^python@3.9.0
...
```

### Information on your system

spack debug report
```
* **Spack:** 0.16.0-804-2cfe349828
* **Python:** 3.6.8
* **Platform:** cray-centos8-aarch64
* **Concretizer:** original
```
spack arch
```
cray-centos8-aarch64
```

### Additional information

* [spack-build-env.txt](https://github.com/spack/spack/files/5848169/spack-build-env.txt)
* [spack-build-out.txt](https://github.com/spack/spack/files/5848170/spack-build-out.txt)

@adamjstewart @aweits 
### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by adamjstewart** [comment]
> Spack compiler must be run from Spack! Input 'SPACK_ENV_PATH' is missing.

As I mentioned in the other issue, Bazel is hacked using `bazelruleclassprovider-0.25.patch` to allow Spack's environment variables to propagate to the build environment. So using a non-Spack-installed Bazel doesn't work at the moment. The last time we talked to the Bazel developers they suggested writing a toolchain for Spack so that we don't need to hack anything. But I don't know much about Bazel and haven't had the time to look into this. For now, Spack-installed Bazel should work.

**Comment by aweits** [comment]
Can we have a look at the bazel build failure? If we get that straightened out things should be *much* easier to address.

**Comment by adamjstewart** [comment]
> Can we have a look at the bazel build failure? 

Please do. The school semester is about to start for me, so I definitely won't have time to look into this in the next few months, but I would be willing to test a PR that does Spack + Bazel the right way.

**Comment by samcom12** [comment]
@aweits ..  attaching latest build logs. 


[spack-build-env.txt](https://github.com/spack/spack/files/6596832/spack-build-env.txt) 
[spack-build-out.txt](https://github.com/spack/spack/files/6596833/spack-build-out.txt) 
[spack-build-01-bootstrap-out.txt](https://github.com/spack/spack/files/6596830/spack-build-01-bootstrap-out.txt)

**Comment by aweits** [comment]
> @aweits .. attaching latest build logs.
> 
> [spack-build-env.txt](https://github.com/spack/spack/files/6596832/spack-build-env.txt)
> [spack-build-out.txt](https://github.com/spack/spack/files/6596833/spack-build-out.txt)
> [spack-build-01-bootstrap-out.txt](https://github.com/spack/spack/files/6596830/spack-build-01-bootstrap-out.txt)

Have you tried using gcc < 11? It seems gcc has gotten more particular about certain headers in version 11...

**Comment by frobnitzem** [comment]
I have heard that bazel is failing to build with versions of java newer than 11.  This is a problem because there is no special `depends_on('java@11')` call-out that prevents spack from always trying (unsuccessfully) from building bazel with newer java versions.

**Comment by adamjstewart** [comment]
Bazel's Java support seems really finicky. I've gotten certain versions of Java to work on one machine but not others. I haven't yet determined what versions are supported, if any.

**Comment by adamjstewart** [comment]
> I have heard that bazel is failing to build with versions of java newer than 11. This is a problem because there is no special `depends_on('java@11')` call-out that prevents spack from always trying (unsuccessfully) from building bazel with newer java versions.

Added in #31402

**Comment by adamjstewart** [comment]
There are also a couple of PRs to fix GCC 11 support now: #30655 #28548

**Comment by adamjstewart** [comment]
External bazel is not currently supported, unless @LydDeb or someone else would like to add support for it. Spack-installed bazel and TF should now work, even for GCC 13+. Let me know if you're still having issues building TF.
